command/agent/command.go

```diff
@@ -134,10 +165,26 @@ func (c *Command) readConfig() *Config {
 		return config
 	}
 
-	// Check that we have a data-dir
-	if config.DataDir == "" {
+	// Check for valid modes.
+	if config.Server.Enabled && config.Client.Enabled {
+		c.Ui.Error("To run both as a server and client, use -dev mode.")
```

Comment:
Do these need to be mutually exclusive? I've used this mode a few times before to see how the client behaves and persists data in a bare minimum set-up. If there's a reason we should actually enforce this then this looks good.